src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,188 @@
+     * @param target the index of task
+     * @throws WrongMessageException potential exception
+     */
+    public void unmark(int target) throws WrongMessageException {
+        Task willUnmark = tasks.getTask(target - 1);
+        willUnmark.nodone();
+        System.out.println("You undone this task:\n"
+                + willUnmark.toString());
+    }
+
+    /**
+     * show the list of task.
+     *
+     * @throws WrongMessageException potential exception
+     */
+    public void showList() throws WrongMessageException {
+        System.out.println("Your list is as following");
+        for (int i = 1; i <= tasks.getSize(); i++) {
+            Task temp = tasks.getTask(i);
+            System.out.println((i) + "." + temp.toString());
+        }
+    }
+
+    /**
+     * delete the targeted task.
+     *
+     * @param str the index of task
+     * @throws WrongMessageException potential exception
+     */
+    public void delete(String str) throws WrongMessageException {
+        String[] temp = str.split(" ");
+        int key3 = Integer.decode(temp[1]);
+        System.out.println("ok I will delete the task " + tasks.deleteTask(key3 - 1) + " right now!");
+        System.out.println("now you have " + tasks.getSize() + " tasks in the list");
+    }
+
+    /**
+     * return the list of task on given date.
+     *
+     * @param localDate the given date
+     */
+    public void getOnDate(LocalDate localDate) {
+        List<Task> shortList = tasks.getTaskList().stream().filter(task -> task.isOnDate(localDate))
+                .collect(Collectors.toList());
+        int i = 0;
+        System.out.println("Hey, these are what you need to do on this date: "
+                + localDate.format(DateTimeFormatter.ofPattern("MMMM d yyyy")));
+        for (Task t : shortList) {
+            System.out.println((i + 1) + "." + t);
+            i++;
+        }
+    }
+
+    /**
+     * return the list of tasks containing given String.
+     *
+     * @param require the given string
+     */
+    public void search(String require) {
+        List<Task> shortList = tasks.getTaskList().stream()
+                .filter(task -> task.hasThis(require))
+                .collect(Collectors.toList());
+        int i = 0;
+        if (shortList.size() > 0) {
+            System.out.println("search result of " + require + " are here: ");
+            for (Task t : shortList) {
+                System.out.println((i + 1) + "." + t);
+                i++;
+            }
+        } else {
+            System.out.println("Sorry, We cannot find those for you");
+        }
+    }
+
+    /**
+     * Run the chatbot, Life is good.
+     */
+    public void run() {
+        ui.greet();
+
+        while (true) {
+            String str = ui.requirement();
+            try {
+                if (str.startsWith("todo")) {
+                    tasks.addTask(Parser.todo(str));
+                    System.out.println("Now you have " + tasks.getSize() + " task(s)");
```

Comment:
Perhaps you can abstract this line into another method? So that there is less repetitive lines in your code.